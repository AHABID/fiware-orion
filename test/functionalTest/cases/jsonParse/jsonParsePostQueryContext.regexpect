1: ***********************************
HTTP/1.1 200 OK
Content-Length: 94
Content-Type: application/json
Date: REGEX(.*)

{
    "errorCode": {
        "code": "404",
        "reasonPhrase": "No context element found"
    }
}
2: ***********************************
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: REGEX(.*)

{
    "duration": "PT1M",
    "registrationId": "REGEX([0-9a-f]{24})"
}
3: ***********************************
HTTP/1.1 200 OK
Content-Length: 94
Content-Type: application/json
Date: REGEX(.*)

{
    "errorCode": {
        "code": "404",
        "reasonPhrase": "No context element found"
    }
}
4: ***********************************
