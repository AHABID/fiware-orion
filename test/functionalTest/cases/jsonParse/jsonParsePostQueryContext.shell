echo "1: ***********************************"
url="/NGSI10/queryContext"
payload='{
  "entities": [
    {
      "type": "Room",
      "isPattern": "false",
      "id": "ConferenceRoom"
    },
    {
      "type": "Room",
      "isPattern": "false",
      "id": "OfficeRoom"
    }
  ],
  "attributes": [
    "temperature",
    "occupancy",
    "lightstatus"
  ],
  "restriction": {
    "attributeExpression" : "Attribute Expression",
    "scopes": [
      {
        "type" : "st1",
        "value" : "sv1"
      },
      {
        "type" : "st2",
        "value" : "sv2"
      }
    ]
  }
}'
curlJson ${url} "${payload}"

echo "2: ***********************************"
url="/NGSI9/registerContext"
payload='{
  "contextRegistrations": [
    {
      "entities": [
        {
          "type": "Room",
          "isPattern": "false",
          "id": "ConferenceRoom"
        },
        {
          "type": "Room",
          "isPattern": "false",
          "id": "OfficeRoom"
        }
      ],
      "attributes": [
        {
          "name": "temperature",
          "type": "degree",
          "isDomain": "false",
          "metadatas": [
            {
              "name": "ID",
              "type": "string",
              "value": "1110"
            }
          ]
        }
      ],
      "metadatas": [
        {
          "name": "ID",
          "type": "string",
          "value": "2212"
        }
      ],
      "providingApplication": "http://192.168.100.1:70/application"
    }
  ],
  "duration": "PT1M"
}'
curlJson ${url} "${payload}"

echo "3: ***********************************"
url="/NGSI10/queryContext"
payload='{
  "entities": [
    {
      "type": "Room",
      "isPattern": "false",
      "id": "ConferenceRoom"
    },
    {
      "type": "Room",
      "isPattern": "false",
      "id": "OfficeRoom"
    }
  ],
  "attributes": [
    "temperature",
    "occupancy",
    "lightstatus"
  ],
  "restriction": {
    "attributeExpression" : "Attribute Expression",
    "scopes": [
      {
        "type" : "st1",
        "value" : "sv1"
      },
      {
        "type" : "st2",
        "value" : "sv2"
      }
    ]
  }
}'
curlJson ${url} "${payload}"

echo "4: ***********************************"
