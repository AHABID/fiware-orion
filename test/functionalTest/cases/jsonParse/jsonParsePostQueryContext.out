1: ***********************************
HTTP/1.1 200 OK
Content-Length: 94
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:18:17 GMT

{
    "errorCode": {
        "code": "404", 
        "reasonPhrase": "No context element found"
    }
}
2: ***********************************
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:18:17 GMT

{
    "duration": "PT1M", 
    "registrationId": "538c3349631c6e0b37d1b1ef"
}
3: ***********************************
HTTP/1.1 200 OK
Content-Length: 94
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:18:17 GMT

{
    "errorCode": {
        "code": "404", 
        "reasonPhrase": "No context element found"
    }
}
4: ***********************************
